### PR TITLE
Add continuous integration on Windows

### LINKED ISSUE #2715: Continuous integration for Windows
We should try AppVeyor as a replacement for travis for Windows continuous integration.

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

This PR adds a file .appveyor.yml for continuous integration on windows with MSVC 2013. It works quite like travis CI but with a Windows environment. 3D Demos have been disabled because of troubles with compiling libQGLViewer.

## Release Management
* Issue(s) solved (if any): fix #2715

### CODE REVIEW

**Comment by MaelRL** [comment]
Cancelling / restarting jobs is restricted to the owner of the PR on AppVeyor ?

**Comment by maxGimeno** [comment]
I think it is restricted to @lrineau actually, cause I cannot do it either.

**Comment by lrineau** [comment]
Please rebase on `master`, so we can see if `master` is still tested OK by AppVeyor.

**Comment by maxGimeno** [comment]
I found it a little too complicated to rebase it after the recent (massive) changes in the travis architecture, so I just hard reset on master and adapted all the files.

**Comment by lrineau** [comment]
You are right. That was probably the best way to do it!

**Comment by lrineau** [comment]
That does not work correctly. In all jobs, the CMake configuration stops with:
```
-- This program requires the CGAL library, and will not be compiled.
```
and then nothing is compiled. See for example, `for test/Mesh_3/`: https://ci.appveyor.com/project/CGAL/cgal/build/1.0.17/job/ab3e526swq8plk2o#L4211

File: .travis/build_package.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (lrineau) [inline .travis/build_package.sh]: The shebang `#!/bin/bash` has to be the first line. It cannot be the second line.

File: .travis/build_package.sh [hunk 2]
```diff
@@ -8,10 +9,10 @@ function build_examples {
   mkdir -p build-travis
   cd build-travis
 if [ $IS_WINDOWS ]; then
-  cmake -DCGAL_DIR="/usr/local/lib/cmake/CGAL" -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS}" -G "Visual Studio 12 2013 Win64" ..
+  cmake -DCGAL_DIR="C:\projects\cgal\build" -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS}" -G "Visual Studio 12 2013 Win64" ..
   cmake --build .
 else
-  cmake -DCGAL_DIR="/usr/local/lib/cmake/CGAL" -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS}" ..
+  cmake -DCGAL_DIR="C:\projects\cgal\build" -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS}" ..
```
Comment (lrineau) [inline .travis/build_package.sh]: In the `else`, `[ $IS_WINDOWS]` returns false, so you should assume the platform is Linux (Travis), and the `CGAL_DIR` should be the path for CGAL on Travis.

File: .travis/build_package.sh [hunk 3]
```diff
@@ -55,10 +56,10 @@ function build_demo {
...
+    cmake -DCGAL_DIR="C:\projects\cgal\build" -DQt5_DIR="/opt/qt55/lib/cmake/Qt5" -DQt5Svg_DIR="/opt/qt55/lib/cmake/Qt5Svg" -DQt5OpenGL_DIR="/opt/qt55/lib/cmake/Qt5OpenGL" -DCGAL_DONT_OVERRIDE_CMAKE_FLAGS:BOOL=ON -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS} ${EXTRA_CXX_FLAGS}" -G "Visual Studio 12 2013 Win64" ..
     cmake --build .
   else
-    cmake -DCGAL_DIR="/usr/local/lib/cmake/CGAL" -DQt5_DIR="/opt/qt55/lib/cmake/Qt5" -DQt5Svg_DIR="/opt/qt55/lib/cmake/Qt5Svg" -DQt5OpenGL_DIR="/opt/qt55/lib/cmake/Qt5OpenGL" -DCGAL_DONT_OVERRIDE_CMAKE_FLAGS:BOOL=ON -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS} ${EXTRA_CXX_FLAGS}" ..
+    cmake -DCGAL_DIR="C:\projects\cgal\build" -DQt5_DIR="/opt/qt55/lib/cmake/Qt5" -DQt5Svg_DIR="/opt/qt55/lib/cmake/Qt5Svg" -DQt5OpenGL_DIR="/opt/qt55/lib/cmake/Qt5OpenGL" -DCGAL_DONT_OVERRIDE_CMAKE_FLAGS:BOOL=ON -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS} ${EXTRA_CXX_FLAGS}" ..
```
Comment (lrineau) [inline .travis/build_package.sh]: Same here for the `else`

**Comment by lrineau** [comment]
This change is [Reviewable](https://reviewable.io/reviews/cgal/cgal/2722)

**Comment by lrineau** [comment]
The AppVeyor build failed:
https://ci.appveyor.com/project/CGAL/cgal/build/1.0.44/job/ih54jyu0ub1fmec1#L811

**Comment by maxGimeno** [comment]
After a quick discusison, we decided to drop this solution. Appveyor takes far too long to perform a testsuite, it won't be of any use after all.

**Pull request closed without merging**